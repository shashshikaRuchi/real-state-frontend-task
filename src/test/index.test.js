import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import LogTable from "../components/logTable";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

describe("Employee Data", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LogTable />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Should all labels render properly", () => {
    render(
      <BrowserRouter>
        <LogTable />
      </BrowserRouter>
    );

    const getEmployeeLabel = screen.getByRole("heading", {
      name: /log id/i,
    });
    expect(getEmployeeLabel).toBeInTheDocument();
    const actionTypeLable = screen.getByRole("heading", {
      name: /application type/i,
    });

    expect(actionTypeLable).toBeInTheDocument();
    const applicationLable = screen.getByRole("heading", {
      name: /application id/i,
    });

    expect(applicationLable).toBeInTheDocument();

    const fromDate = screen.getByRole("heading", { name: /action type/i });

    expect(fromDate).toBeInTheDocument();

    const toDate = screen.getByRole("heading", { name: /from date/i });

    expect(toDate).toBeInTheDocument();

    const applicationId = screen.getByRole("heading", { name: /to date/i });

    expect(applicationId).toBeInTheDocument();

    const searchLoggerButton = screen.getByRole("button", {
      name: /search logger/i,
    });

    expect(searchLoggerButton).toBeInTheDocument();
  });
  test("Should all input render correctely", () => {
    const { container } = render(
      <BrowserRouter>
        <LogTable />
      </BrowserRouter>
    );
    const employeeInput = screen.getByRole("spinbutton", { name: /logid/i });
    expect(employeeInput).toBeInTheDocument();
    const applicationIdInput = screen.getByRole("combobox", {
      name: /applicationtype/i,
    });

    expect(applicationIdInput).toBeInTheDocument();
    const applicationTypeSelect = screen.getByRole("spinbutton", {
      name: /applicationid/i,
    });

    expect(applicationTypeSelect).toBeInTheDocument();
    const actionTypeSelect = screen.getByRole("combobox", {
      name: /actiontype/i,
    });

    expect(actionTypeSelect).toBeInTheDocument();
    const fromDateSelector = container.querySelector(`input[name="fromDate"]`);

    expect(fromDateSelector).toBeInTheDocument();
    const toDateSelector = container.querySelector(`input[name="toDate"]`);

    expect(toDateSelector).toBeInTheDocument();
  });
});
