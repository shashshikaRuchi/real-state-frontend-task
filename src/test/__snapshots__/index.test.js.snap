// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Data renders correctly 1`] = `
<div
  className="App"
  data-testid="EmployeeTable"
>
  <div
    className="App"
    data-testid="EmployeeFilterField"
  >
    <div
      className="rowStyleContainer"
    >
      <div
        className="columnStyle"
      >
        <h4>
          Log ID
        </h4>
        <input
          aria-label="LogId"
          data-testid="LogId"
          name="LogId"
          onChange={[Function]}
          placeholder="Log Id"
          type="number"
          value=""
        />
      </div>
      <div
        className="columnStyle"
      >
        <h4>
          Application type
        </h4>
        <select
          aria-label="ApplicationType"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Select ApplicationType
          </option>
        </select>
      </div>
      <div
        className="columnStyle"
      >
        <h4>
          Application ID
        </h4>
        <input
          aria-label="ApplicationId"
          data-testid="ApplicationId"
          name="ApplicationId"
          onChange={[Function]}
          placeholder="application ID"
          type="number"
          value=""
        />
      </div>
      <div
        className="columnStyle"
      >
        <h4>
          Action type
        </h4>
        <select
          aria-label="ActionType"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Select ActionType
          </option>
        </select>
      </div>
      <div
        className="columnStyle"
      >
        <h4>
          From Date
        </h4>
        <div
          className="rowStyleContainer"
        >
          <input
            aria-label="fromDate"
            data-testid="fromDate"
            name="fromDate"
            onChange={[Function]}
            placeholder="Select Date"
            type="date"
            value="Select Date"
          />
        </div>
      </div>
      <div
        className="columnStyle"
      >
        <h4>
          To Date
        </h4>
        <div
          className="rowStyleContainer"
        >
          <input
            aria-label="toDate"
            data-testid="toDate"
            name="toDate"
            onChange={[Function]}
            type="date"
          />
        </div>
      </div>
      <button
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "blue",
            "height": "50px",
            "marginLeft": "10px",
            "marginTop": "20px",
          }
        }
        type="submit"
      >
        Search Logger
      </button>
    </div>
  </div>
  <table>
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          Log ID
        </th>
        <th
          onClick={[Function]}
        >
          Application Type
        </th>
        <th
          onClick={[Function]}
        >
          Application ID
        </th>
        <th
          onClick={[Function]}
        >
          Action Type
        </th>
        <th>
          Action Details
        </th>
        <th
          onClick={[Function]}
        >
          Date:Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div>
    <ul
      className="pagination justify-content-center"
    >
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          &lt;&lt;
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          &gt;&gt;
        </button>
      </li>
    </ul>
  </div>
</div>
`;
